### PR TITLE
Feature friend model #52

### PR DESCRIPTION
**Pull request by m3tron**
- Get all friends
- Add and Remove friends
- Get suggested users that aren't your friend

### CODE REVIEW

File: client/src/pages/container/DashboardDefault.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import { Container, Typography } from "@material-ui/core";
 import FriendListCard from "../presentational/FriendListCard";
-import FriendListDialog from "../presentational/FriendListDialog";
+import FriendListDialog from "../presentational/FriendlistDialog";
```
Comment (k-awe-some) [inline client/src/pages/container/DashboardDefault.jsx]: I changed this one to `FriendListDialog` so we can be more consistent in naming thru out the app (with a capital `L`)

**Review by k-awe-some — COMMENTED** [review]
Looks good overall!

File: client/src/pages/container/DashboardDefault.jsx (hunk 1 above)
Comment (m3tron) [reply client/src/pages/container/DashboardDefault.jsx]: the actual component has a lower case l that's why i changed it i'll just change the component name

**Review by chungleee — COMMENTED** [review]
looks good and ready to merge!

**Pull request merged**